Apromore-Extras/APMLogModule/src/main/java/org/apromore/apmlog/DefaultChartDataCollection.java

```diff
@@ -92,41 +74,33 @@ public DefaultChartDataCollection(APMLog apmLog) {
         for(int i = 0; i< apmLog.getTraceList().size() ; i++) {
 
             ATrace aTrace = apmLog.getTraceList().get(i);
-
             List<AEvent> aEventList = aTrace.getEventList();
 
-            if(aEventList.size() > 0) {
-                String caseName = aTrace.getCaseId();
+            caseDurations.add( aEventList.size() == 0 ? 0 : aTrace.getDuration() );
+            ttlProcessTimes.add( aEventList.size() == 0 ? 0 : aTrace.getTotalProcessingTime() );
+            avgProcessTimes.add( aEventList.size() == 0 ? 0 : aTrace.getAverageProcessingTime() );
+            maxProcessTimes.add( aEventList.size() == 0 ? 0 : aTrace.getMaxProcessingTime() );
+            ttlWaitTimes.add( aEventList.size() == 0 ? 0 : aTrace.getTotalWaitingTime() );
+            avgWaitTimes.add( aEventList.size() == 0 ? 0 : aTrace.getAverageWaitingTime() );
+            maxWaitTimes.add( aEventList.size() == 0 ? 0 : aTrace.getMaxWaitingTime() );
+            caseUtils.add( aEventList.size() == 0 ? 0 : aTrace.getCaseUtilization() );
```

Comment:
I feel like this would be tidier if you moved the repeated aEventList.size() test outside, something like this:
```
if (aEventList.size() != 0) {
    caseDurations.add(aTrace.getDuration());
    ttlProcessTimes.add(aTrace.getTotalProcessingTime());
    // etc
}
```